dialer.go

```diff
@@ -43,25 +44,43 @@ func (d *Dialer) Dial(network string, address string) (net.Conn, error) {
 // (*net.Dialer).Dialcontext documentation at
 // https://golang.org/pkg/net/#Dialer.DialContext.
 func (d *Dialer) DialContext(ctx context.Context, network string, address string) (net.Conn, error) {
-	if host, _ := splitHostPort(address); len(host) != 0 && net.ParseIP(host) == nil {
-		addrs, err := d.resolver().LookupService(ctx, host)
+	host, _ := splitHostPort(address)
+	resolve := len(host) != 0 && net.ParseIP(host) == nil
+	attempt := 0
+
+	resolver := d.resolver()
+	dialer := &net.Dialer{
+		Timeout:       d.Timeout,
+		Deadline:      d.Deadline,
+		LocalAddr:     d.LocalAddr,
+		DualStack:     d.DualStack,
+		FallbackDelay: d.FallbackDelay,
+		KeepAlive:     d.KeepAlive,
+	}
+
+	if !resolve {
+		return dialer.DialContext(ctx, network, address)
+	}
+
+	for {
+		addrs, err := resolver.LookupService(ctx, host)
 		if err != nil {
 			return nil, err
 		}
 		if len(addrs) == 0 {
 			return nil, fmt.Errorf("no addresses returned by the resolver for %s", host)
 		}
 		address = addrs[0].Addr.String()
-	}
+		conn, err := dialer.DialContext(ctx, network, address)
 
-	return (&net.Dialer{
-		Timeout:       d.Timeout,
-		Deadline:      d.Deadline,
-		LocalAddr:     d.LocalAddr,
-		DualStack:     d.DualStack,
-		FallbackDelay: d.FallbackDelay,
-		KeepAlive:     d.KeepAlive,
-	}).DialContext(ctx, network, address)
+		if err != nil && attempt < 10 && resolver.Blacklist != nil {
+			resolver.Blacklist.Blacklist(addrs[0].Addr, time.Now().Add(d.blacklistTTL()))
```

Comment:
Should we be incrementing the index for `addrs` when looking through the SRV response?  Maybe I don't fully understand the context of this PR.

Although we do another SRV query for each loop it seems.  Probably cached locally but still seems odd.

Also wouldn't we want a sleep somewhere in this loop?

Further, we could check the `err.Temorary` interface for network connections.

Lemme know if I'm way off here in what this PR is trying to accomplish.